File: 30daysOFcode_Php/day17.php
<?php

class Calculator
{
    public function __construct()
    {

    }

    public function power($n, $p)
    {
        if ($n < 0 || $p < 0) {
            throw new Exception('n and p should be non-negative');
        }
        return pow($n, $p);
    }
}

$myCalculator = new Calculator();
$T = intval(fgets(STDIN));
while ($T-- > 0) {
    list($n, $p) = explode(" ", trim(fgets(STDIN)));
    try {
        $ans = $myCalculator->power($n, $p);
        echo $ans . "\n";
    } catch (Exception $e) {
        echo $e->getMessage() . "\n";
    }
}

File: 30daysOFcode_Swift/day 17.swift
class Calculator {
    // Start of function power
    func power(n: Int, p: Int) throws -> Int {
        // Add your code here
        if (n < 0 || p < 0) {
            throw RangeError.NotInRange("n and p should be non-negative")
        }
        var result = 1
        for _ in 0..<p  {
            result *= n
        }
        return result
    } // End of function power
}
